app/_layout.tsx

```diff
@@ -1,58 +1,91 @@
 
 export {
-  // Catch any errors thrown by the Layout component.
-  ErrorBoundary,
-} from 'expo-router';
+    // Catch any errors thrown by the Layout component.
+    ErrorBoundary,
+} from 'expo-router'
 
 export const unstable_settings = {
-  // Ensure that reloading on `/modal` keeps a back button present.
-  initialRouteName: '(tabs)',
-};
+    // Ensure that reloading on `/modal` keeps a back button present.
+    initialRouteName: '(tabs)',
+}
 
 // Prevent the splash screen from auto-hiding before asset loading is complete.
-SplashScreen.preventAutoHideAsync();
+SplashScreen.preventAutoHideAsync()
 
 export default function RootLayout() {
-  const [loaded, error] = useFonts({
-    SpaceMono: require('../assets/fonts/SpaceMono-Regular.ttf'),
-    ...FontAwesome.font,
-  });
-
-  // Expo Router uses Error Boundaries to catch errors in the navigation tree.
-  useEffect(() => {
-    if (error) throw error;
-  }, [error]);
-
-  useEffect(() => {
-    if (loaded) {
-      SplashScreen.hideAsync();
-    }
-  }, [loaded]);
+    const [loaded, error] = useFonts({
+        SpaceMono: require('../assets/fonts/SpaceMono-Regular.ttf'),
+        ...FontAwesome.font,
+    })
+
+    // Expo Router uses Error Boundaries to catch errors in the navigation tree.
+    useEffect(() => {
+        if (error) throw error
+    }, [error])
+
+    useEffect(() => {
+        if (loaded) {
+            SplashScreen.hideAsync()
+        }
+    }, [loaded])
 
-  if (!loaded) {
-    return null;
-  }
+    if (!loaded) {
+        return null
+    }
 
-  return <RootLayoutNav />;
+    return <RootLayoutNav />
 }
 
 function RootLayoutNav() {
-  const colorScheme = useColorScheme();
-
-  return (
-    <ThemeProvider value={colorScheme === 'dark' ? DarkTheme : DefaultTheme}>
-      <Stack>
-        <Stack.Screen name="(tabs)" options={{ headerShown: false }} />
-        <Stack.Screen name="modal" options={{ presentation: 'modal' }} />
-      </Stack>
-    </ThemeProvider>
-  );
+    const colorScheme = useColorScheme()
+
+    return (
+        <ThemeProvider
+            value={colorScheme === 'dark' ? DarkTheme : DefaultTheme}
+        >
+            <Stack
+                screenOptions={{
+                    headerStyle: {
+                        backgroundColor: Colors.primary,
+                    },
+                    headerTitle: () => (
+                        <Image
+                            source={require('../assets/images/logo.png')}
+                            style={styles.image}
+                        />
+                    ),
+                }}
+            >
+                <Stack.Screen name="(tabs)" options={{ headerShown: true }} />
+                <Stack.Screen
+                    name="modal"
+                    options={{ presentation: 'modal' }}
+                />
+            </Stack>
+        </ThemeProvider>
+    )
 }
+
+
+const styles = StyleSheet.create({
+    image: {
+        width: 50,
+        height: 50,
+    },
+})
```

Comment:
<img width="339" alt="Screenshot 2024-05-06 at 20 41 03" src="https://github.com/Spolka-Zlo/D17Map-Mobile/assets/51704650/3a34c68b-94da-4b37-a7ea-96660e92ecef">
end of file line missing for some reason